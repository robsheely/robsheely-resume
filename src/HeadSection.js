import React from "react";
import PropTypes from "prop-types";

import Typography from "@material-ui/core/Typography";
import { withStyles } from "@material-ui/core";

import photo from "./media/profilePicture.jpg";

const styles = theme => ({
  root: {
    marginTop: -16,
    marginLeft: -16,
    marginRight: -16,
    textAlign: "center"
  },
  data: {
    marginLeft: 30,
    marginRight: 30
  },
  header: {
    padding: 26,
    marginBottom: 16,
    backgroundColor: theme.palette.primary.main,
    display: "flex",
    justifyContent: "space-between"
  },
  text: {
    color: theme.palette.secondary[ "600" ],
    marginBottom: 10
  },
  headerText: {
    color: theme.palette.primary.contrastText
  },
  headerTextDiv: {
    marginLeft: -100,
    marginTop: 4,
    marginBottom: -8,
    flexGrow: 1,
    flexDirection: "column",
    display: "flex",
    justifyContent: "space-between"
  },
  summaryText: {
    textAlign: "left"
  },
  h3: {
    fontWeight: 500
  }
});

const HeadSection = props => {
  const { classes, name, contact, head, subhead, summary } = props;

  return (
    <div
      id="header-section"
      data-testid="header-section"
      className={classes.root}
    >
      <div className={classes.header}>
        <img
          id="header-section-image"
          data-testid="header-section-image"
          className={classes.image}
          src={photo}
          alt="Rob Sheely"
          title="Rob Sheely"
          height="100"
        />
        <div className={classes.headerTextDiv}>
          <Typography
            id="header-section-name"
            data-testid="header-section-name"
            variant="h3"
            classes={{
              root: classes.headerText,
              h3: classes.h3
            }}>
            {name}
          </Typography>
          <Typography
            id="header-section-contact"
            data-testid="header-section-contact"
            variant="body1"
            className={classes.headerText}>
            {contact}
          </Typography>
        </div>
      </div>
      <div className={classes.data}>
        <Typography
          id="header-section-head"
          data-testid="header-section-head"
          variant="h5"
          className={classes.text}>
          {head}
        </Typography>
        <Typography
          id="header-section-subhead"
          data-testid="header-section-subhead"
          variant="h6"
          className={classes.text}>
          {subhead}
        </Typography>
        <Typography
          id="header-section-summary"
          data-testid="header-section-summary"
          variant="body2"
          className={classes.summaryText}>
          {summary}
        </Typography>
      </div>
    </div>
  );
};

HeadSection.propTypes = {
  classes: PropTypes.object.isRequired,
  name: PropTypes.string.isRequired,
  contact: PropTypes.string.isRequired,
  head: PropTypes.string.isRequired,
  subhead: PropTypes.string.isRequired,
  summary: PropTypes.string.isRequired
};

export default withStyles(styles)(HeadSection);
