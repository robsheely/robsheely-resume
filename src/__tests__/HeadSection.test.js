import React from "react";
import { render, cleanup } from "react-testing-library";

import HeadSection from "../HeadSection";

const MOCK_DATA = {
  name: "Name",
  contact: "Phone | Address | Email",
  head: "Good Programmer",
  subhead: "Makes Good Programs",
  summary: "A really good programmer who makes good programs."
};

describe("HeadSection", () => {
  afterEach(cleanup);

  it("Passes smoke test", () => {
    const { container } = render(<HeadSection {...MOCK_DATA}/>);

    expect(container).toMatchSnapshot();
  });

  it("Contains expected children & content", () => {
    const { queryByTestId } = render(<HeadSection {...MOCK_DATA}/>);

    expect(queryByTestId("header-section")).not.toBeNull();
    expect(queryByTestId("header-section-name").textContent).toBe(MOCK_DATA.name);
    expect(queryByTestId("header-section-contact").textContent).toBe(MOCK_DATA.contact);
    expect(queryByTestId("header-section-head").textContent).toBe(MOCK_DATA.head);
    expect(queryByTestId("header-section-subhead").textContent).toBe(MOCK_DATA.subhead);
    expect(queryByTestId("header-section-summary").textContent).toBe(MOCK_DATA.summary);
  });
});
