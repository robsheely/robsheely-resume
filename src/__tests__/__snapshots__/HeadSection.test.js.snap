// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeadSection Passes smoke test 1`] = `
<div>
  <div
    class="HeadSection-root-1"
    data-testid="header-section"
    id="header-section"
  >
    <div
      class="HeadSection-header-3"
    >
      <img
        alt="Rob Sheely"
        data-testid="header-section-image"
        height="100"
        id="header-section-image"
        src="profilePicture.jpg"
        title="Rob Sheely"
      />
      <div
        class="HeadSection-headerTextDiv-6"
      >
        <h3
          class="MuiTypography-root-9 HeadSection-headerText-5 MuiTypography-h3-23 HeadSection-h3-8"
          data-testid="header-section-name"
          id="header-section-name"
        >
          Name
        </h3>
        <p
          class="MuiTypography-root-9 MuiTypography-body1-18 HeadSection-headerText-5"
          data-testid="header-section-contact"
          id="header-section-contact"
        >
          Phone | Address | Email
        </p>
      </div>
    </div>
    <div
      class="HeadSection-data-2"
    >
      <h5
        class="MuiTypography-root-9 MuiTypography-h5-25 HeadSection-text-4"
        data-testid="header-section-head"
        id="header-section-head"
      >
        Good Programmer
      </h5>
      <h6
        class="MuiTypography-root-9 MuiTypography-h6-26 HeadSection-text-4"
        data-testid="header-section-subhead"
        id="header-section-subhead"
      >
        Makes Good Programs
      </h6>
      <p
        class="MuiTypography-root-9 MuiTypography-body2-17 HeadSection-summaryText-7"
        data-testid="header-section-summary"
        id="header-section-summary"
      >
        A really good programmer who makes good programs.
      </p>
    </div>
  </div>
</div>
`;
